### PR TITLE
[ADD] account_payment_multicurrency module

### PR DESCRIPTION
**Pull request by vrenaville**
This module add the possibility to see the amount in currency and to be able to select partial paid invoice

### CODE REVIEW

**Comment by vrenaville** [comment]
@pedrobaeza @sbidoul do you have any comments on this PR

Thanks for your help,

Vincent

**Comment by sbidoul** [comment]
@vrenaville I think you should consider making your module depend on account_banking_payment_export which provides a hook _prepare_payment_line. That should help you avoid copying the whole create_payment method.

Actually, I wonder if this issue is not fixed in account_banking_payment_export, can you check?

**Comment by sbidoul** [comment]
If not, perhaps it's even easier to put your fix in account_banking_payment_export directly.

**Comment by vrenaville** [comment]
@sbidoul thanks for your comment, I don't understand on the previous PR, I will test it and if it fit the need I will see to improve the view if required.

**Comment by fclementic2c** [comment]
Hi Stephane,

I agree with your comment but it is difficult to do so in v7 since we already have several modules (swiss SEPA payement for instance) which are not base on this account_banking_payment_export and in production with several customers. 

But we will keep in mind your remark for v8

**Comment by sbidoul** [comment]
@fclementic2c I understand.

Since this new module overrides create_payment without calling super() I'm concerned it will break other installs using account_banking_payment_export (which also overrides create_payment without calling super()...). 

The root of the problem is in the offical account_payment, of course.

However, introducing a conflicting module in OCA/bank-payment may create confusion.

**Comment by vrenaville** [comment]
Hello,

@sbidoul  thanks for the review, I have add a super instead a complete rewrite of the function

Vincent

**Comment by vrenaville** [comment]
@sbidoul I have discuss with @guewen I will improve my module, on monday.
Vincent

**Comment by vrenaville** [comment]
@sbidoul I have just improve my code to have correct value assignment directly to payment line instead of search it.

Vincent

File: account_payment_multicurrency_extension/payment.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    def create_payment(self, cr, uid, ids, context=None):
+        """
+        We add is_multi_currency tag to be able set
+        'amount_currency': line.amount_residual_currency
+        instead of
+        'amount_currency': line.amount_to_pay
+        To be compliant with multi currency
+        Allready corrected in V8 but will not be corrected in V7
+        """
```
Comment (guewen) [inline account_payment_multicurrency_extension/payment.py:36]: context could be `None`, add a safeguard:

```
if context is None:
    context = {}
```

File: account_payment_multicurrency_extension/payment.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            ids,
+            context=context)
+
+
+class payment_line(orm.Model):
+    _inherit = 'payment.line'
+
+    def create(self, cr, uid, vals, context=None):
+        """In case of multi currency
+        we use amount_residual_currency instead of amount_to_pay"""
```
Comment (guewen) [inline account_payment_multicurrency_extension/payment.py:53]: context could be None, add a safeguard:

if context is None:
    context = {}

**Comment by vrenaville** [comment]
@guewen Thanks for the review.

I have correct the module according to your comment

Vincent

File: account_payment_multicurrency_extension/payment.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    _inherit = 'payment.order.create'
+
+    def create_payment(self, cr, uid, ids, context=None):
+        """
+        We add is_multi_currency tag to be able set
+        'amount_currency': line.amount_residual_currency
+        instead of
+        'amount_currency': line.amount_to_pay
+        To be compliant with multi currency
+        Allready corrected in V8 but will not be corrected in V7
```
Comment (gurneyalex) [inline account_payment_multicurrency_extension/payment.py]: s/Allready/Already/

File: account_payment_multicurrency_extension/payment.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        """
+        if context is None:
+            context = {}
+        context.update({'is_multi_currency': True})
```
Comment (gurneyalex) [inline account_payment_multicurrency_extension/payment.py]: do we really want the modified context to be seen by the caller?

the usual idiom is to make a copy and modify the copy.

**Comment by fclementic2c** [comment]
functionnal test :+1:

**Comment by vrenaville** [comment]
@gurneyalex I have made the fix according to your comment

**Comment by guewen** [comment]
Red due to pylint warnings in other modules :+1:

**Pull request merged**